tests/ui/abi/statics/static-mut-foreign.rs

```diff
@@ -19,15 +19,22 @@ fn static_bound_set(a: &'static mut c_int) {
 
 unsafe fn run() {
     assert_eq!(rust_dbg_static_mut, 3);
+    //~^ WARN creating a shared reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.